### ISSUE TITLE
Instead of opening renpy and passing environment variables, connect to a running renpy

### ISSUE THREAD

**Issue by brunoais** [furudean/vscode-renpy-warp#28]
A bigger limitation of the current implementation of `vscode-renpy-warp ` is that it ends up monopolizing renpy (or, at least) require being the one starting renpy, otherwise, it won't run or work at all, even if just because the client doesn't know how to connect to the server.

I have some ideas how it might be done differently but I'll need to test them first.

Note: This issue is being made because some initial commenting on that was made here: https://github.com/furudean/vscode-renpy-warp/issues/24#issuecomment-2253510864

As stated, I will be doing multiple PoC myself to understand options and flaws. I'll report back with options when I have them.

**Comment by brunoais** [comment]
Hey. I have an initial working proposal for how it might work.
There's a file with the port available just outside the game directory (code for that is in <kbd>_websocket_addon.py:36</kbd>). Clients are supposed to read that file to get the port to connect to.
Every new instance of the game will open a different port.
The file contains the port of the latest opened instance of the game.
The last instance of the game deletes the file (ofc, if it crashes, segfault style or the process is killed by the OS, the file will stay).
Because this needs to work in windows, I had to make design decisions to accommodate its imposed limitations. I hope I didn't forget any.

I purposely didn't try to avoid race conditions with the file which stores the port number to keep the code more simple.

I included an example.py which tries to exemplify how to use the API. Hopefully it's clear enough and the API is simple enough.

The POC:
[00_devwebsocket.rpe.zip](https://github.com/user-attachments/files/16431762/00_devwebsocket.rpe.zip)
